### PR TITLE
`Development`: Add window query and dynamic filtering

### PR DESCRIPTION
**Pull request by MoSchmidt**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://confluence.aet.cit.tum.de/spaces/AP/pages/257786006/Language+Guidelines+Client).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311714/PR+Guidelines).

#### Server

- [x] **Important**: I implemented the changes with a [very good performance](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311718/Performance+Guidelines) and prevented too many (unnecessary) and too complex database calls.
- [x] I **strictly** followed the principle of **data economy** for all database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311877/Server+Guidelines).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

The ApplicationCarousel requires both standard paginated results and an easily navigable “window” of records around a focal application.
- Performance: Fetching only the subset of records needed by the carousel reduces payload size and database load.
- Consistency: Centralizing sorting and filtering logic on the server ensures that client-side filters are applied uniformly.

### Description

# Description
This PR introduces two new REST endpoints in `ApplicationEvaluationController` to support the client’s ApplicationCarousel component:

1. **`GET /application-details`**  
   - Returns a paginated list of application evaluation details.  
   - Accepts:
     - `OffsetPageDTO offsetPageDTO` (page number, page size)
     - `SortDTO sortDto` (sorting criteria)
     - `FilterDTO filterDto` (dynamic filtering parameters)  
   - Delegates to `applicationEvaluationService.getApplicationsDetails(...)`.

2. **`GET /application-details/window`**  
   - Returns a “sliding window” of application details centered around a specified application ID.  
   - Accepts:
     - `UUID applicationId` (center of the window)
     - `int windowSize` (must be a positive odd integer)
     - `SortDTO sortDto`  
     - `FilterDTO filterDto`  
   - Delegates to `applicationEvaluationService.getApplicationsDetailsWindow(...)`.

> **Temporary note:** Both endpoints still use a hard-coded `ResearchGroup` stub (`00000000-0000-0000-0000-000000000001`) until authentication is wired up to populate it from the logged-in user.

### Steps for Testing

Only through postman available yet:

Get the window centered around the application:

`http://localhost:8080/api/evaluation/application-details/window?sortBy=rating&direction=DESC&windowSize=7&applicationId=d0b84bf6-e639-4e41-8d03-e2863af24e66`

Get paginated list of applications:

`http://localhost:8080/api/evaluation/application-details?sortBy=rating&direction=DESC&offset=0&limit=10`

For both requests you can add fiters. They have to follow this pattern filters%5B<fieldToFilter>%5D=<value1,value2>

Example: `http://localhost:8080/api/evaluation/application-details?sortBy=rating&direction=DESC&offset=0&limit=10&filters%5Bstate%5D=IN_REVIEW&filters%5Bjob.jobId%5D=<jobId>`

Note: the %5B and %5D are just "[" and "]" encoded because Tomcat does not support those as raw strings in the URL

**Checklist:**
### Checklist for Testers

- [x] Validate that the **paginated list endpoint** returns the correct number of records (matching the requested `limit`).
- [x] Verify that the **window endpoint** returns the correct window size (matching the requested `windowSize`), and that the results are correctly centered around the specified `applicationId`.
- [ ] Check that **sorting** (e.g., by `rating` or other fields) works as expected in both endpoints.
- [x] Verify that **dynamic filters** (e.g., `filters[state]=IN_REVIEW,ACCEPTED`) are correctly applied and that filtered results are accurate.
- [x] Ensure that the **sliding window still returns a valid window** when filters are applied (i.e., the centered application appears in the results and the window is correctly sized).
- [x] Test edge cases for `windowSize` (e.g., small odd numbers like 1 or 3, larger odd numbers) and verify correct behavior.
- [x] Confirm that invalid `windowSize` values (e.g., negative numbers, even numbers) result in appropriate error responses.
- [x] Confirm that invalid or non-existent `applicationId` values result in appropriate error handling.

#### Code Review

- [ ] Code Review 1

#### Manual Tests

- [ ] Test 1

### CODE REVIEW

File: src/main/java/de/tum/cit/aet/evaluation/dto/ApplicationEvaluationDetailDTO.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package de.tum.cit.aet.evaluation.dto;
+
+import de.tum.cit.aet.application.domain.Application;
+import de.tum.cit.aet.usermanagement.domain.Applicant;
+import java.util.UUID;
+
+//TODO define additional fields
```
Comment (MoSchmidt) [inline src/main/java/de/tum/cit/aet/evaluation/dto/ApplicationEvaluationDetailDTO.java:7]: @Douglas-Sarah add the fields for details page here

**Comment by MoSchmidt** [comment]
Thank you all for the feedback. @Douglas-Sarah, @ani-ib thanks for finding the bug with sorting for the last name. It was in fact a typo that is solved now. It should work with "applicant.lastName".
I agree that sorting for the state would be great but we are storing it as a string (char) in our database. So we could sort by alphabetical order but I think this would not be the result users expect. To get a custom order would only we possible with extra effort, so I left it out for now. But this is defeintly an improvement for the future.

@Douglas-Sarah as far as I know, we do not have an appliedAt field in the database, so I can not sort for it. Or am I missing something here?

File: src/main/java/de/tum/cit/aet/evaluation/web/ApplicationEvaluationResource.java [hunk 2]
```diff
@@ -23,22 +22,72 @@ public class ApplicationEvaluationResource {
...
+     */
+    @GetMapping("/application-details/window")
+    public ResponseEntity<ApplicationEvaluationDetailListDTO> getApplicationsDetailsWindow(
+        @RequestParam UUID applicationId,
+        @RequestParam int windowSize,
+        @ParameterObject @ModelAttribute SortDTO sortDto,
+        @ParameterObject @ModelAttribute FilterDTO filterDto
+    ) {
+        //TODO this will be removed when the ResearchGroup can be accessed through the authenticated user
+        ResearchGroup researchGroup = new ResearchGroup();
```
[the remark this answers is not in the document]
Comment (MoSchmidt) [reply src/main/java/de/tum/cit/aet/evaluation/web/ApplicationEvaluationResource.java:87]: What do you mean by parameter verification? If you mean validation then this would be part of a future PR

File: src/main/java/de/tum/cit/aet/evaluation/dto/ApplicationEvaluationDetailDTO.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    LocalDateTime appliedAt
+) {
+    /**
+     * Creates an {@link ApplicationEvaluationDetailDTO} from the given {@link Application} entity.
+     *
+     * @param application the {@link Application} entity
+     * @return a new {@link ApplicationEvaluationDetailDTO} populated from the application data
+     */
+    public static ApplicationEvaluationDetailDTO fromApplication(Application application) {
+        Applicant applicant = application.getApplicant();
```
[the remark this answers is not in the document]
Comment (MoSchmidt) [reply src/main/java/de/tum/cit/aet/evaluation/dto/ApplicationEvaluationDetailDTO.java:31]: Those are response DTOs so no need for null checks

**Review by Douglas-Sarah — APPROVED** [review]
Thank you for fixing the issue! It works great now.
You are right- we don't yet have a column `appliedAt` yet. I thought we did because we use createdAt for our DTOs and call them appliedAt^^
For now, that is good enough. I suggest we should extend the database in the next sprint, because we have the saveAsDraft feature, which leads to the problem that those applications have a different apply-date. 
It also makes sense that sorting according to the state column is more complicated 👍  I am fine with adding that feature at a later point

**Review by ani-ib — APPROVED** [review]
Thanks for fixing the issue, I've tested it locally once again and sorting by professor.lastName works as expected now. Other than that, code LGTM 👍

File: src/main/java/de/tum/cit/aet/evaluation/web/ApplicationEvaluationResource.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (marc-fett) [reply src/main/java/de/tum/cit/aet/evaluation/web/ApplicationEvaluationResource.java:87]: I mean things like 
```
if (windowSize == null || windowSize <= 0 || (windowSize % 2) != 1) {
    throw new IllegalArgumentException("Window size must be a positive and odd integer");
}
```
from getApplicationsDetailsWindow

**Review by az108 — APPROVED** [review]
LGTM 👍 Thanks for adding the comments
